### DISCUSSION TITLE
Bottom Sheet Dialog 높이를 컨텐츠가 차지하는 영역만큼 키우려면 어떻게 해야 할까요?

### DISCUSSION THREAD

**Discussion by djyunz** [holixfactory/bottomsheetdialog-compose#34] [Q&A] [answered]
일반 사이즈의 emulator 에서는 크게 문제(숨겨진 부분의 크기 작음)가 되지 않는데

7.6 Fold-in 사이즈의 펼쳐진 큰 화면에서만 문제가 되는 것 같네요.

지금은 PeekHeight.Auto 로 되어 있지만 실제로 컨텐츠 영역만큼 커지는 것은 아니고 
일정 크기 만큼만 커지고 스크롤 되는 형태가 되는 것 같습니다. 즉, 일부 하단 영역은 숨겨진 상태로 노출됩니다.

컨텐츠 영역은 아래와 같은 구조를 가지고 있습니다.

```
Column {
  ...
LazyVerticalGrid {
  ...
}
```

peakHeight 를 컨텐츠 영역 크기 만큼 직접 세팅을 해야 할까요?

**Comment by djyunz** [comment, marked as answer]
제가 Bottom Sheet Dialog 의 구조를 잘 이해하지 못했던 것 같네요.

BottomSheetBehaviorProperties.State 의 기본 값이 

`Collapsed`

인데,  실제로는 

`HalfExpanded` 

상태로 로 처음 화면에 노출되는 것 같습니다.

아래와 같이 완전 펼쳐진 상태 (Expanded) 로 설정하니 원하는 결과를 얻었네요.

`behaviorProperties = BottomSheetBehaviorProperties(
                state = BottomSheetBehaviorProperties.State.Expanded
            )`

**Reply by workspace** [reply]
네. 맞습니다! 본 라이브러리는 material2 BottomSheetDialog를 감싸서 compose에서 사용할 수 있도록 하고 있습니다. 그에 따라 BottomSheetDialog의 설정값에 대한 설명이 좀 미비했던 것 같은데 알맞는 설정값을 찾으셨다니 다행입니다! 데모앱에서 모든 property를 테스트 해볼 수 있도록 해뒀으니 알맞은 설정값을 찾을 때 도움이 되셨으면 합니다👍
